app/helpers/application_helper.rb

```diff
@@ -433,7 +433,13 @@ def hidden_spinner(text = '', options = {})
   end
 
   def hosts_count(resource_name = controller.resource_name)
-    @hosts_count ||= Host::Managed.reorder('').authorized.group("#{resource_name}_id").count
+    # If we are on /organizations or /locations, this allows to display the
+    # count for hosts not in the current organization & location.
+    if ['organization', 'location'].include? resource_name
+      hosts_scope = Host::Managed.reorder('').unscoped
```

Comment:
No? This would only display the count, once you click on the number to be directed to the search, it will not display you anything you cannot see.